packages/frontend/component/src/lit-react/lit-portal/lit-portal.tsx

```diff
@@ -98,48 +91,32 @@ export const useLitPortalFactory = () => {
             ? elementOrFactory()
             : elementOrFactory;
         return createLitPortalAnchor(event => {
-          const { name, target } = event;
-          const id = target.portalId;
-
-          if (name === 'connectedCallback') {
-            setPortals(portals => [
-              ...portals,
-              {
-                id,
-                portal: ReactDOM.createPortal(element, target, id),
-              },
-            ]);
-            return;
-          }
-
-          if (name === 'disconnectedCallback') {
-            setPortals(portals => portals.filter(p => p.id !== id));
-            return;
-          }
-
-          const prevId = event.previousPortalId;
-
-          // Ignores first `willUpdate`
-          if (!prevId) {
-            return;
-          }
-
           setPortals(portals => {
-            const portal = portals.find(p => p.id === prevId);
-            if (!portal) return [...portals];
-
-            // Updates `ID`
-            // Used for portal queries in `disconnectedCallback`
-            portal.id = id;
-
-            // Re-rendering
-            //  true: `inline link`
-            //  false: `pdf embed view`
-            if (rerendering) {
-              portal.portal = ReactDOM.createPortal(element, target, id);
+            const { name, target } = event;
+            const id = target.portalId;
+            let newPortals = portals;
+            switch (name) {
+              case 'willUpdate':
+                if (
+                  (!rerendering && target.hasUpdated) ||
+                  target.disconnected
```

Comment:
use `target.isConnected`

Reply:
no, using disconnected is intended. An element will recieve willUpdate even when element is not counnected.

Reply:
use `!target.isConnected`